### PR TITLE
Added configuration for typing speed unit, removed alwaysShowCPM

### LINKED ISSUE #4491: Replace 'always show cpm' with 'units'
Add settings such as: wpm, cpm, wps, cps

### PR DESCRIPTION
**Pull request by fehmer**
### Description

Remove config `alwaysShowCPM` and replace with `typingSpeedUnit`. Supported units are `wpm`, `cpm`, `wps` and `cps`.

Closes #4491

### CODE REVIEW

File: frontend/src/ts/account/pb-tables.ts [hunk 1]
```diff
@@ -127,25 +127,30 @@ function buildPbHtml(
 ): string {
   let retval = "";
   let dateText = "";
-  const multiplier = Config.alwaysShowCPM ? 5 : 1;
   const modeString = `${mode2} ${mode === "time" ? "seconds" : "words"}`;
-  const wpmCpm = Config.alwaysShowCPM ? "cpm" : "wpm";
+  const wpmCpm = Config.typingSpeedUnit;
```
Comment (Miodec) [inline frontend/src/ts/account/pb-tables.ts:131]: Rename this variable

File: frontend/src/ts/account/pb-tables.ts [hunk 2]
```diff
@@ -127,25 +127,30 @@ function buildPbHtml(
...
   try {
     const pbData = (pbs[mode][mode2] ?? []).sort((a, b) => b.wpm - a.wpm)[0];
     const date = new Date(pbData.timestamp);
     if (pbData.timestamp) {
       dateText = format(date, "dd MMM yyyy");
     }
 
-    let wpmString: number | string = pbData.wpm * multiplier;
+    let wpmString: number | string = Misc.convertTypingSpeed(
```
Comment (Miodec) [inline frontend/src/ts/account/pb-tables.ts:139]: Rename to `speedString` or `stringWIthUnit`

File: frontend/src/ts/constants/default-config.ts [hunk 3]
```diff
@@ -73,7 +73,7 @@ export default <MonkeyTypes.Config>{
   minWpm: "off",
   minWpmCustomSpeed: 100,
   highlightMode: "letter",
-  alwaysShowCPM: false,
+  typingSpeedUnit: "wpm", //TODO: change to wpm
```
Comment (Miodec) [inline frontend/src/ts/constants/default-config.ts:76]: What does this todo mean?

File: frontend/src/ts/utils/misc.ts [hunk 4]
```diff
@@ -491,6 +491,44 @@ export function roundTo2(num: number): number {
...
+}
+
+export function convertTypingSpeedWithUnitSuffix(
+  unit: MonkeyTypes.TypingSpeedUnit,
+  wpm: number
+): string {
+  return roundTo2(convertTypingSpeed(unit, wpm)).toFixed(2) + " " + unit;
+}
+
+export function translateTypingSpeed(
```
Comment (Miodec) [inline frontend/src/ts/utils/misc.ts:517]: Maybe rename to something like `getFullSpeedUnitString` ?

**Review by Miodec — CHANGES_REQUESTED** [review]
Also, make sure to add `alwaysShowCPM` to the `replaceLegacyValues` in `config.ts`

File: frontend/src/ts/account/pb-tables.ts (hunk 1 above)
Comment (fehmer) [reply frontend/src/ts/account/pb-tables.ts]: done

File: frontend/src/ts/account/pb-tables.ts (hunk 2 above)
Comment (fehmer) [reply frontend/src/ts/account/pb-tables.ts]: done

File: frontend/src/ts/utils/misc.ts (hunk 4 above)
Comment (fehmer) [reply frontend/src/ts/utils/misc.ts]: done

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/5716538325

File: frontend/src/ts/pages/account.ts [hunk 5]
```diff
@@ -713,22 +721,30 @@ async function fillContent(): Promise<void> {
...
+        j += bucketSize
+      ) {
         histogramChartDataBucketed.push({ x: j, y: 0 });
-        labels.push(`${j} - ${j + 9}`);
+        labels.push(`${j} - ${j + bucketSize - 1}`);
       }
     }
   }
 
+  console.log(labels);
```
Comment (fehmer) [inline frontend/src/ts/pages/account.ts:746]: TODO remove

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/5749204337

**Comment by Miodec** [comment]
image

Something is still wrong with the account graph scale, one result is not visible

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/5749954903

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/5750029152

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/5750029152

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/5751271990

**Pull request merged**